### PR TITLE
Paying for College - Move expandable code to flyout code

### PR DESCRIPTION
**Pull request by mistergone**
A change in `cf-expandables` broke customization in the Financial Path tool, and @anselmbradford generously gave us guidance on how to fix this so that we weren't dependent on the `cf-expandables` pattern (which should not be customized). 

## Changes
- remove references to `cf-expandables`
- Change class names
- Update JS, including adding 

## How to test this PR
1. Generally, the "expandables" in the Covering Your Costs > Make a Plan section of the app should look better!
2. All features of the application should function as before.

## Notes and todos

-

## Checklist

- [ ] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

#### Browser testing

Visually tested in the following supported browsers:
- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge 18 (the last Edge prior to it switching to Chromium)
- [ ] Internet Explorer 11 and 8 (via emulation in 11's dev tools)
- [ ] Safari on iOS
- [ ] Chrome on Android

#### Accessibility

- [ ] Keyboard friendly (navigable with tab, space, enter, arrow keys, etc.)
- [ ] Screen reader friendly
- [ ] Does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)

#### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Does not introduce new lint warnings
- [ ] Flexible from small to large screens

### CODE REVIEW

**Comment by mistergone** [comment]
Sorry for the delays - this should be ready for primetime!

File: cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less [hunk 1]
```diff
@@ -112,11 +131,38 @@
...
+
+    &.u-is-animating {
+      overflow: hidden;
+    }
+  }
+
+  .financial-item_label p {
+    font-size: unit(5px / @base-font-size-px, em);
+  }
 }
```
Comment (anselmbradford) [inline cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less:150]: Does it still work if you remove all this?

File: cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less [hunk 2]
```diff
@@ -112,11 +131,38 @@
...
+    .u-clearfix();
+
+    &__transition {
+      transition: max-height @flyout__transition-speed ease-in-out;
+    }
+
+    &.u-is-animating {
+      overflow: hidden;
+    }
+  }
```
Comment (anselmbradford) [inline cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less:145]: Does it still work if you remove all this?

File: cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less [hunk 3]
```diff
@@ -89,10 +99,19 @@
...
+      .o-costs-group_label {
         flex: 1;
+        text-align: left;
+        flex-grow: 1;
       }
     }
+
+    .o-costs-group_link .o-costs-group_cue {
+      text-align: right;
+      color: #0072ce;
```
Comment (anselmbradford) [inline cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less:111]: Should be `color: @pacific`, yea?

File: cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less [hunk 4]
```diff
@@ -89,10 +99,19 @@
     .flex-container {
       display: flex;
 
-      .o-expandable_label {
+      .o-costs-group_label {
         flex: 1;
+        text-align: left;
```
Comment (anselmbradford) [inline cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less:104]: What's making the `text-align: left` necessary? This isn't the default?

File: cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less [hunk 5]
```diff
@@ -65,22 +72,25 @@
     }
   }
 
-  &.o-expandable_header {
-    &[aria-expanded='false'] {
-      .o-expandable_cue-close {
+  &.o-costs-group_target {
```
Comment (anselmbradford) [inline cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less:75]: `o-expandable_target` is gone, as it was redundant to the header. We should probably follow suit here and have it be `o-costs-group_header`.

File: cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-flyout.html [hunk 6]
```diff
@@ -26,19 +26,20 @@
...
    ========================================================================== #}
 
-{% macro expandable(value) %}
-<div class="o-expandable
-            o-expandable__padded">
-    <button class="o-expandable_header
+{% macro flyout(value) %}
+<div class="o-costs-group" data-js-hook="behavior_flyout-menu">
+    <button class="o-costs-group_header
+                   o-costs-group_target
```
Comment (anselmbradford) [inline cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-flyout.html:32]: `o-expandable_target` is gone. You shouldn't need both `o-costs-group_header` and `o-costs-group_target`. Just merge them together into `o-costs-group_header`.

File: cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-flyout.html [hunk 7]
```diff
@@ -26,19 +26,20 @@
...
+                   o-costs-group_target__collapsed
```
Comment (anselmbradford) [inline cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-flyout.html:33]: Is this a modifier for it to be collapsed on load? Isn't that the default?

File: cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-flyout.html [hunk 8]
```diff
@@ -56,7 +57,7 @@
             </span>
         {% endif %}
     </button>
-    <div class="o-expandable_content">
+    <div class="o-costs-group_content" data-js-hook="behavior_flyout-menu_content">
```
Comment (anselmbradford) [inline cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-flyout.html:60]: The data JS hook being added [here](https://github.com/cfpb/consumerfinance.gov/pull/7686/files#diff-9df3aa6bcb4547f45882fd3e86e56c0c5cbbf268451efe80abf415ef3effe6c4R37-R40), so hardcoding it into the markup is not necessary/redundant.

File: cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-flyout.html [hunk 9]
```diff
@@ -26,19 +26,20 @@
 
    ========================================================================== #}
 
-{% macro expandable(value) %}
-<div class="o-expandable
-            o-expandable__padded">
-    <button class="o-expandable_header
+{% macro flyout(value) %}
+<div class="o-costs-group" data-js-hook="behavior_flyout-menu">
```
Comment (anselmbradford) [inline cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-flyout.html:30]: The data JS hook being added [here](https://github.com/cfpb/consumerfinance.gov/pull/7686/files#diff-9df3aa6bcb4547f45882fd3e86e56c0c5cbbf268451efe80abf415ef3effe6c4R37-R40), so hardcoding it into the markup is not necessary/redundant.

File: cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-flyout.html [hunk 10]
```diff
@@ -26,19 +26,20 @@
...
+    <button class="o-costs-group_header
+                   o-costs-group_target
+                   o-costs-group_target__collapsed
                    financial-item
-                   financial-item__expandable
-                   {{ ('financial-item__' + value.status | safe ) if value.status else ''}}"
+                   financial-item__flyout
+                   {{ ('financial-item__' + value.status | safe ) if value.status else ''}}" data-js-hook="behavior_flyout-menu_trigger"
```
Comment (anselmbradford) [inline cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-flyout.html:36]: The data JS hook being added [here](https://github.com/cfpb/consumerfinance.gov/pull/7686/files#diff-9df3aa6bcb4547f45882fd3e86e56c0c5cbbf268451efe80abf415ef3effe6c4R37-R40), so hardcoding it into the markup is not necessary/redundant.

File: cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less (hunk 2 above)
Comment (mistergone) [reply cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less:145]: Yep! 👍

File: cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less [hunk 11]
```diff
@@ -112,11 +131,38 @@
...
-  .financial-item {
-    margin-top: unit(15px / @base-font-size-px, em);
+  .flyout {
+      border-bottom: 1px solid @flyout__border;
+      &:first-child {
+        border-top: 1px solid @flyout__border;
+    }
+  }
+
+  .o-costs-group-content {
```
[the remark this answers is not in the document]
Comment (mistergone) [reply cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less:160]: Ah, cool. I always mess these up.

File: cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less [hunk 12]
```diff
@@ -112,11 +131,38 @@
...
+
+  .financial-item_label p {
+    font-size: unit(5px / @base-font-size-px, em);
+  }
 }
 
 .financial-item-group {
-  .financial-item {
-    margin-top: unit(15px / @base-font-size-px, em);
+  .flyout {
```
[the remark this answers is not in the document]
Comment (mistergone) [reply cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less:153]: Yep, and that fixes a style error I just noticed!

File: cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less [hunk 13]
```diff
@@ -89,10 +99,19 @@
...
-      .o-expandable_label {
+      .o-costs-group_label {
         flex: 1;
+        text-align: left;
+        flex-grow: 1;
       }
     }
+
+    .o-costs-group_link .o-costs-group_cue {
+      text-align: right;
```
[the remark this answers is not in the document]
Comment (mistergone) [reply cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less:110]: There were other problems with this little bit, but I think I've clarified them (in code I'll push shortly).

File: cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less (hunk 4 above)
Comment (mistergone) [reply cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less:104]: Oh wait, no, `<BUTTON>` centers text, which is why it's there, but there may be a better way to do this...

File: cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less (hunk 5 above)
Comment (mistergone) [reply cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less:75]: Sounds good, I'll work that in.

File: cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-flyout.html (hunk 6 above)
Comment (mistergone) [reply cfgov/paying_for_college/jinja2/paying-for-college/college-cost-blocks/macros/financial-item-flyout.html:32]: 👍

File: cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less (hunk 4 above)
Comment (anselmbradford) [reply cfgov/unprocessed/apps/paying-for-college/css/college-costs/financial-item.less]: Oh right, from https://github.com/cfpb/design-system/blob/main/packages/cfpb-expandables/src/expandable.less#L55 … I agree. I wonder if there's a better way than override button defaults.

**Review by wpears — APPROVED** [review]
Neat!

**Pull request merged**